fix(action): escape special characters from the prompt

jacobtomlinson opened the pull request:
<!--
Filling out the information below can facilitate the review/merge of the pull request (PR).
-->

## What is the motivation for this pull request?

The action fails when you include special characters in your prompt.

## What is the current behavior?

The action fails when you include special characters in your prompt.

## What is the new behavior?

The prompt is escaped so that this doesn't happen.

## Checklist:

<!--
Feel free to remove any item that is irrelevant to your changes.
To check an item, place an "x" in the box like so: `- [x] Tests`
-->

- [ ] [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Tests
- [ ] Documentation

<!--
Any other comments? Thank you for contributing!
-->

remarkablemark commented on action.yml:
```diff
@@ -33,7 +33,7 @@ runs:
         {
           echo "response<<$EOF"
-          ollama run $MODEL "$PROMPT"
+          ollama run $MODEL "${PROMPT@Q}"
```
What does the `@Q` do?

jacobtomlinson replied:
It escapes characters like double quotes that would break the prompt when included directly.

remarkablemark replied:
Thanks for the explanation!

remarkablemark reviewed (changes requested):
Thanks for opening the PR! It looks like the [macos test](https://github.com/ai-action/ollama-action/actions/runs/14710269083/job/41295532105?pr=4) is failing:

```
${PROMPT@Q}: bad substitution
```

Could I also get your help in adding a test?

jacobtomlinson commented:
It looks like macOS has bach 4.2 and the `@Q` expansion was added in 4.4. I've updated the PR to use `printf` instead and I've also added a test. The use of backticks and other special characters in my prompt was what was causing things to fail for me originally.

remarkablemark commented:
> It looks like macOS has bash 4.2 and the `@Q` expansion was added in 4.4. I've updated the PR to use `printf` instead and I've also added a test. The use of backticks and other special characters in my prompt was what was causing things to fail for me originally.

Thanks for the update, looks good to me!
